sdk/samples/iot/pnp/pnp_protocol.c

```diff
@@ -375,7 +390,8 @@ az_result pnp_process_device_twin_message(
     }
     else if (jr.token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT)
     {
-      if (az_result_failed(az_json_reader_skip_children(&jr)))
+      rc = az_json_reader_skip_children(&jr);
+      if (az_result_failed(rc))
```

Comment:
Unrelated: If the result is not getting logged here OR returned, do we need to set it?

Alternatively, should the `rc` be returned instead of a custom `az_result`?

Reply:
Good question.  So I ran into the same boolean comparison bug we have in the vs compiler even if the function call was embedded inside an az_result_failed() check.

For that reason, I am getting the rc code first and then passing it to az_result_failed().  I think the call as it was (before my change here) reads fine, but didn't know how far to take this restructure due to the current bug.  What do you think?

Reply:
> Alternatively, should the `rc` be returned instead of a custom `az_result`?

I don't know why we are explicitly returning `AZ_ERROR_UNEXPECTED_CHAR` here rather than just returning the `az_result` from the `az_json_reader_skip_children` call. Another reason that could fail is `AZ_ERROR_UNEXPECTED_END`. How do you know which error to return?

Reply:
this is all being cleared up in my next PR